libs/full/performance_counters/include/hpx/performance_counters/counters_fwd.hpp

```diff
@@ -191,17 +191,50 @@ namespace hpx { namespace performance_counters {
     ///////////////////////////////////////////////////////////////////////////
     // Status and error codes used by the functions related to
     // performance counters.
-    enum counter_status
+    enum class counter_status
     {
-        status_valid_data,         // No error occurred, data is valid
-        status_new_data,           // Data is valid and different from last call
-        status_invalid_data,       // Some error occurred, data is not value
-        status_already_defined,    // The type or instance already has been defined
-        status_counter_unknown,         // The counter instance is unknown
-        status_counter_type_unknown,    // The counter type is unknown
-        status_generic_error            // A unknown error occurred
+        valid_data,         // No error occurred, data is valid
+        new_data,           // Data is valid and different from last call
+        invalid_data,       // Some error occurred, data is not value
+        already_defined,    // The type or instance already has been defined
+        counter_unknown,    // The counter instance is unknown
+        counter_type_unknown,    // The counter type is unknown
+        generic_error            // A unknown error occurred
     };
 
+    inline constexpr std::ostream& operator<<(
```

Comment:
@K-ballo Any ideas on how to tackle the `all paths through this function will call itself` error here? If `counter_status` was just a class and not an enum class I would just provide a `friend` overload for the operator inside the class body. But now...?

Reply:
You need to cast `rhs` to an integer, then it will do what you want.

Reply:
`static_cast` or just `int`? and why?